clients/vault/src/oracle/mod.rs

```diff
@@ -0,0 +1,25 @@
+#![allow(dead_code)]
+
+mod collector;
+mod constants;
+mod errors;
+mod handler;
+mod storage;
+mod types;
+
+pub use handler::*;
+pub use storage::prepare_directories;
+
+use collector::*;
+use errors::Error;
+use storage::*;
+use types::*;
+
+/// A filter trait to check whether `T` should be processed.
+pub trait FilterWith<T> {
+	/// unique identifier of this filter
+	fn id(&self) -> u32;
```

Comment:
By the way, could we (the oracle library code) also just take this over so the user of the trait does not have to provide an id? Can we have some global counter variable, maybe, that is just incremented for every new filter? Otherwise the user would have to keep track of the IDs he already used which might be annoying if he creates filters at different places.

Reply:
hhm. yeah, we can do that. Alright. 
But how to delete them (SHOULD we need to delete, that's another question too) later on?

Reply:
Ahhhh true, that's why we needed the user to provide it in the first place... 😅 
Well no, let's keep it then. But can we maybe make the ID a bit generic so that either an integer or a &str can be used? It might be more convenient if you can provide some kind of name for the filter and not just a number. Or if you don't want to make it generic, I'd prefer the &str over number as id.

Reply:
gotcha, i'll change the datatype.